add nethserver branding and sample apps toggle placeholder

ksat-design opened the pull request:
I've made changes and reverted them back to your version then I readied the nethserver branding and added the toggle for sample apps (it's non functional at the moment)

Commit:
Copy local theme, apps, and backgrounds into the repository

Commit:
Copy local theme, apps, and backgrounds into the repository

Commit:
Update custom.css

Commit:
changed nethbackground location

changed nethbackground to theme and set it via custom.css

Commit:
Delete imageroot/state/llng/backgrounds/nethbackground.png

Commit:
add favicon

Commit:
Rename favicon_multi.ico to favicon.ico

Commit:
Create set-branding

Commit:
chmod set-branding

Commit:
toch set-branding

Commit:
added nethbranding and chmod

Commit:
Update lemonldapng-app.service

added ExecStartPost=/usr/local/bin/runagent set-nethbranding

Commit:
Update set-nethbranding

changed to bash script for testing

Commit:
Delete imageroot/bin/set-branding

Commit:
Update set-nethbranding

changed to python

Commit:
Update set-nethbranding

added cmd to remove option to register new account

Commit:
Create update-sample-apps-settings

Commit:
Update update-sample-apps-settings

chmod update-sample-apps-settings

Commit:
Update lemonldapng-app.service

added execstartpost for update-sample-apps-settings

Commit:
Update lemonldapng-app.service

Commit:
Update set-nethbranding

Commit:
add nethbackground

change custom.css

Commit:
Update Settings.vue

test to get toggle for sample apps

Commit:
Update Settings.vue

second attempt

Commit:
Update translation.json

Commit:
Update translation.json

Commit:
Update update-sample-apps-settings

Commit:
Update 10configure_environment_vars

add sample apps

Commit:
Update validate-input.json

sample

Commit:
Update 20read

sample

Commit:
Update validate-output.json

sample

Commit:
Update translation.json

sample

Commit:
Update translation.json

Commit:
Update translation.json

Commit:
Update 20read

Commit:
Merge pull request #1 from ksat-design/ksat-design-patch-3

Ksat design patch 3

Commit:
Update update-sample-apps-settings

test 5

Commit:
Update update-sample-apps-settings

Commit:
Update update-sample-apps-settings

test6

Commit:
Update update-sample-apps-settings

Commit:
Update update-sample-apps-settings

Commit:
Update update-sample-apps-settings

Commit:
Update README.md

Commit:
Update Settings.vue

Reverted to base

Commit:
Update translation.json

Reverted to base

Commit:
Update 10configure_environment_vars

reverted to base

Commit:
Update validate-input.json

Reverted to base

Commit:
Update 20read

revert to base

Commit:
Update validate-output.json

revert to base

Commit:
Update validate-output.json

fixed file to correct one

Commit:
Update lemonldapng-app.service

add background

Commit:
Update Settings.vue

test 2 to get toggle

Commit:
Add files via upload

added missing images

Commit:
Update translation.json

updated sample apps flag name and tooltip

Commit:
Update Settings.vue

set sample apps status to enabled by default

Commit:
Update Settings.vue

fixed file

Commit:
Update Settings.vue

stephdl commented:
> and added the toggle for sample apps (it's non functional at the moment)

What is not workable at the moment, the command to remove the sample apps, the logic, the background lemonldapng-cli command structure ?

Commit:
Update Settings.vue

Commit:
Update translation.json

Commit:
Update 10configure_environment_vars

Commit:
Update validate-input.json

Commit:
Update 20read

Commit:
Update validate-output.json

Commit:
Update lemonldapng-app.service

Commit:
Update lemonldapng.service

Commit:
Update set-nethbranding

Commit:
Update update-sample-apps-settings

Commit:
Update lemonldapng-app.service

Commit:
Update set-nethbranding

Commit:
Update lemonldapng-app.service

Commit:
Update update-sample-apps-settings

Commit:
Update README.md

Commit:
Update translation.json

fixed translation for Spanish and added the cda and saml keys

Commit:
Update translation.json

added saml and cda keys

Commit:
Update translation.json

added cda and saml keys

Commit:
Update translation.json

added saml and cda keys

Commit:
Update translation.json

added cda and saml keys

Commit:
Update translation.json

added cda and saml keys

Commit:
Update translation.json

added saml and cda keys

Commit:
Update translation.json

added saml and cda keys

Commit:
Update metadata.json

updated author

Commit:
Update README.md

altered look and included sync script

Commit:
Update README.md

fixed reference

Commit:
Update 06copyenv

to add cda and saml

Commit:
Update 50traefik

    "cda_status": to_bool(renv.get("CDA_STATUS", "False")),
    "saml_status": to_bool(renv.get("SAML_STATUS", "False")),

Commit:
Update 50Traefik

    "cda_status": to_bool(renv.get("CDA_STATUS", "False")),
    "saml_status": to_bool(renv.get("SAML_STATUS", "False")),

stephdl commented on imageroot/actions/restore-module/50traefik:
```diff
@@ -13,10 +13,23 @@ import os
+# Build configuration data from restored environment
+config_data = {
+    "lets_encrypt": to_bool(renv["TRAEFIK_LETS_ENCRYPT"]),
     "host": renv["TRAEFIK_HOST"],
-    "http2https": renv["TRAEFIK_HTTP2HTTPS"] == "True",
+    "http2https": to_bool(renv["TRAEFIK_HTTP2HTTPS"]),
     "ldap_domain": renv["LDAP_DOMAIN"],
-})
+    "cda_status": to_bool(renv.get("CDA_STATUS", "False")),
+    "saml_status": to_bool(renv.get("SAML_STATUS", "False")),
```
== True convert simply when it find the string `True` to a bool value `true`

you should keep it simple

stephdl commented on imageroot/actions/clone-module/50Traefik:
```diff
@@ -5,13 +5,31 @@
+def to_bool(val): return val == "True"
+
+# Build configuration data from restored environment
+config_data = {
+    "lets_encrypt": to_bool(renv["TRAEFIK_LETS_ENCRYPT"]),
+    "host": renv["TRAEFIK_HOST"],
+    "http2https": to_bool(renv["TRAEFIK_HTTP2HTTPS"]),
+    "ldap_domain": renv["LDAP_DOMAIN"],
+    "cda_status": to_bool(renv.get("CDA_STATUS", "False")),
+    "saml_status": to_bool(renv.get("SAML_STATUS", "False")),
+}
```
same as above, keep the == 'True' to convert to a bool value

stephdl commented on imageroot/actions/restore-module/06copyenv:
```diff
@@ -14,11 +14,12 @@ request = json.load(sys.stdin)
-    agent.set_env(evar, original_environment[evar])
-
-agent.dump_env()
+    "TRAEFIK_HOST",
+    "TRAEFIK_HTTP2HTTPS",
+    "TRAEFIK_LETS_ENCRYPT",
+    "LDAP_DOMAIN",
+    "CDA_STATUS",
+    "SAML_STATUS",
+]:
```
the indentation is not set as before, with python sometimes you could get some surprises :p

Commit:
Update 06copyenv

stephdl commented on imageroot/actions/restore-module/06copyenv:
```diff
@@ -23,3 +23,4 @@ for evar in [
 ]:
     if evar in original_environment:
         agent.set_env(evar, original_environment[evar])
+agent.dump_env()
```
this is obsolete now and does not more than a warning in logs, you can remove it safely, sometimes I forgot to remove it
